### DISCUSSION TITLE
If we set ChromeRuntime = true in CefSettings, our initial setting for CachePath is not taking effect

### DISCUSSION THREAD

**Discussion by snkar-hyland** [cefsharp/CefSharp#4899] [General]
Our code for CefSettings is as follows:
string rootCachePath = '<Some folder path under application profile>';
`RootCachePath = rootCachePath,
CachePath = Path.Combine(rootCachePath, <Sub Folder Path>),
......
......
......
......
......
ChromeRuntime = true`

Initially we did not have the ChromeRuntime = true setting and we were getting the browser cache files stored at %TEMP%/RootCache/. After just setting the Chromium Runtime without any other changes, the Cache Files are not getting stored at all (or are getting stored in a different location which we are not aware of) and hence our application cannot take advantage of the cached files resulting in performance impact. Rest of all the browser capabilities in the application is working fine.

Is there anything documented or reported issues which leads to a different behaviour for CachePath if the ChroimeRuntime is set to true?

**Comment by amaitland** [comment]
What version are you using?

> Initially we did not have the ChromeRuntime = true setting

The `Chrome Runtime` is enabled by default in `M126` and will be removed in `M128`.

You're the only person I've seen reporting issues. Load `chrome://version` in a `ChromiumWebBrowser` instance and see what the cache paths look like.

If you are using the latest version then it might be an **upstream** bug.

**Reply by snkar-hyland** [reply]
We have upgraded to the latest version (126.2.180) and found the issue to have been resolved. We also have set the ChromeRuntime flag to false. We might face a similar issue when we upgrade to the version which removes the ChromeRuntime flag and defaults to the Chrome Runtime by default. We will tackle that situation, if and when it arises. Right now the pending issue is resolved with the latest update.

**Comment by mitchcapper** [comment]
@snkar-hyland is the cachepath an immediate child of rootcachepath?  Prior it only had to be any sub directory under rootcachepath but they now require it to be a direct sub directory.  As @amaitland mentioned use version to see where the cache path actually is.
